core/modules/replica/Configuration.h

```diff
@@ -429,75 +430,610 @@ class Configuration: public ConfigurationIFace {
+                          bool isPartitioned,
+                          std::list<SqlColDef> const& columns=std::list<SqlColDef>(),
+                          bool isDirectorTable=false,
+                          std::string const& directorTableKey="objectId",
+                          std::string const& chunkIdColName=lsst::qserv::CHUNK_COLUMN,
+                          std::string const& subChunkIdColName=lsst::qserv::SUB_CHUNK_COLUMN,
+                          std::string const& latitudeColName=std::string(),
+                          std::string const& longitudeColName=std::string());
+
+    /**
+     * Delete an existing table.
+     * @param database The name of an existing database hosting the table.
+     * @param table The name of an existing table to be deleted.
+     * @throw std::invalid_argument If the specified database doesn't exists, or
+     *   if the table doesn't exist, or if either of those parameters are
+     *   the empty strings.
+     */
+    DatabaseInfo deleteTable(std::string const& database,
+                             std::string const& table);
+
+    /**
+     * @param name The name of a worker.
+     * @return 'true' if the specified worker is known to the configuration.
+     */
+    bool isKnownWorker(std::string const& name) const;
+
+    /**
+     * @param name The name of a worker.
+     * @return A worker descriptor.
+     * @throw std::invalid_argument If the specified worker was not found in
+     *   the configuration.
+     */
+    WorkerInfo workerInfo(std::string const& name) const;
+
+    /**
+     * Register a new worker in the Configuration.
+     * @param workerInfo The worker description.
+     * @return A worker descriptor.
+     * @throw std::invalid_argument If the specified worker was not found in
+     *   the configuration.
+     */
+    WorkerInfo addWorker(WorkerInfo const& info);
+
+    /**
+     * Completely remove the specified worker from the Configuration.
+     * @param name The name of a worker affected by the operation.
+     * @throw std::invalid_argument If the specified worker was not found in
+     *   the configuration.
+     */
+    void deleteWorker(std::string const& name);
+
+    /**
+     * Disable the specified worker in the Configuration to exclude it from using
+     * in any subsequent replication operations.
+     * @throw std::invalid_argument If the specified worker was not found in
+     *   the configuration.
+     */
+    WorkerInfo disableWorker(std::string const& name);
+
+    /**
+     * Update parameters of an existing worker in the transient store, and in
+     * the persistent back-end as well (if any is associated with the Configuration object).
+     * @param info The new worker descriptor.
+     * @return An updated worker descriptor.
+     * @throw std::invalid_argument If the specified worker was not found in
+     *   the configuration.
+     */
+    WorkerInfo updateWorker(WorkerInfo const& info);
```

Comment:
This approach simplifies the API as it avoids introducing a bunch of methods for modifying the individual properties of a worker (something some folks complained about a lot in the past). The concurrency isn't an issue since this `updateWorker` is a synchronized method of the class `Configuration` (see `Configuration::_mtx` and `util::Lock` used in the implementations of the relevant public methods dealing with the non-atomic structures.), and updates in the MySQL backend are done within transactions.
The current way of updating the worker parameters is:
```
WorkerInfo info = config->workerInfo(worker);
info.par1 = new-val-of-par1;
info.par2 = mew val-of-par2;
...
config->updateWorker(info);
```

Reply:
Concurrency is always a problem, if you have two separate processes updating database concurrently then transactions don't help you, result will depend on ordering.